### PR TITLE
add more unit test to check the getDeviceforImage function

### PR DESCRIPTION
**Pull request by mgold1234**
Signed-off-by: mgold1234 <mgold@redhat.com>

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/services/images_test.go [hunk 1]
```diff
@@ -1698,31 +1698,49 @@ var _ = Describe("Image Service Test", func() {
...
 				device = []models.Device{
-					{OrgID: common.DefaultOrgID, UUID: faker.UUIDHyphenated(), ImageID: image1.ID},
-					{OrgID: common.DefaultOrgID, UUID: faker.UUIDHyphenated(), ImageID: image1.ID},
+					{OrgID: common.DefaultOrgID, UUID: faker.UUIDHyphenated(), ImageID: image2.ID},
+					{OrgID: common.DefaultOrgID, UUID: faker.UUIDHyphenated(), ImageID: image2.ID},
 				}
 
 			})
 			Context("Get Devices count image", func() {
 				It("should return device count of image1", func() {
```
Comment (tomershinhar) [inline pkg/services/images_test.go:1719]: the name suggests that you would call image1 but image2 is used instead

File: pkg/services/images_test.go [hunk 2]
```diff
@@ -1698,31 +1698,49 @@ var _ = Describe("Image Service Test", func() {
...
 					db.DB.Debug().Create(&device)
-					count, err := service.GetImageDevicesCount(image1.ID)
+					count, err := service.GetImageDevicesCount(image2.ID)
 					Expect(err).To(BeNil())
 					Expect(count).To(Equal(int64(2)))
 
 				})
 				It("should return 0 device of image2", func() {
```
Comment (tomershinhar) [inline pkg/services/images_test.go:1726]: image2 is the one running on the devices, and the test itself correctly calls image1
the name needs to be updated too

File: pkg/services/images_test.go [hunk 3]
```diff
@@ -1698,31 +1698,49 @@ var _ = Describe("Image Service Test", func() {
...
 					db.DB.Debug().Create(&device)
-					count, err := service.GetImageDevicesCount(image2.ID)
+					count, err := service.GetImageDevicesCount(image1.ID)
 					Expect(err).To(BeNil())
 					Expect(count).To(Equal(int64(0)))
 				})
+
+				It("should return 0 device of image2", func() {
```
Comment (tomershinhar) [inline pkg/services/images_test.go:1733]: this name of the test is not matching what it is actually doing

File: pkg/routes/images.go [hunk 4]
```diff
@@ -522,6 +522,8 @@ func GetImageDetailsByID(w http.ResponseWriter, r *http.Request) {
 			imgDetail.UpdateAdded = len(upd[len(upd)-1].PackageDiff.Removed)
 			imgDetail.UpdateRemoved = len(upd[len(upd)-1].PackageDiff.Added)
 			imgDetail.UpdateUpdated = len(upd[len(upd)-1].PackageDiff.Upgraded)
+			imgDetail.Image.TotalDevicesWithImage = upd[len(upd)-1].Image.TotalDevicesWithImage
```
Comment (tomershinhar) [inline pkg/routes/images.go:525]: we might want to add validation for those added attributes in the route test too, just to make sure they are returned correctly

File: pkg/services/images_test.go (hunk 1 above)
Comment (mgold1234) [reply pkg/services/images_test.go:1719]: good catch!!

File: pkg/services/images_test.go (hunk 2 above)
Comment (mgold1234) [reply pkg/services/images_test.go:1726]: thank you for that

File: pkg/services/images_test.go (hunk 3 above)
Comment (mgold1234) [reply pkg/services/images_test.go:1733]: thank you

File: pkg/routes/images.go (hunk 4 above)
Comment (mgold1234) [reply pkg/routes/images.go:525]: I will try

**Review by tomershinhar — DISMISSED** [review]
LGTM

File: pkg/services/images_test.go [hunk 5]
```diff
@@ -1698,31 +1698,49 @@ var _ = Describe("Image Service Test", func() {
...
+						Arch:  "x86_64",
+						InstalledPackages: []models.InstalledPackage{
+							{Name: "vim"},
+						},
+					},
 				}
 				result = db.DB.Create(image2)
 				Expect(result.Error).ToNot(HaveOccurred())
 
 				device = []models.Device{
```
[the remark this answers is not in the document]
Comment (lahavyuv86) [reply pkg/services/images_test.go:1712]: @mgold1234 ^^

[the remark this answers is not in the document]
Comment (mgold1234) [reply pkg/services/images_test.go:1712]: good idea

File: pkg/routes/images.go [hunk 6]
```diff
@@ -522,6 +523,8 @@ func GetImageDetailsByID(w http.ResponseWriter, r *http.Request) {
...
+			imgDetail.Image.TotalPackages = upd[len(upd)-1].Image.TotalPackages
```
Comment (omaciel) [inline pkg/routes/images.go:527]: Zero coverage:

Screenshot 2022-11-28 at 08 50 54

**Review by omaciel — CHANGES_REQUESTED** [review]
Please a single unittest to cover `func GetImageDetailsByID`

**Review by tomershinhar — APPROVED** [review]
LGTM

File: pkg/routes/images.go (hunk 6 above)
Comment (mgold1234) [reply pkg/routes/images.go:526]: @omaciel I added more unit test, can you please look?

**Review by omaciel — APPROVED** [review]
/lgtm

**Pull request merged**